### ISSUE TITLE
Do setExternalUserId and removeExternalUserId respectively sub and unsub the device?

### ISSUE THREAD

**Issue by markoj3s** [OneSignal/react-native-onesignal#1189]
Hi!

My case is as follows:

I have been doing some experiments with 2 devices (1 iOS and 1 Android) by calling from an application the create subscription api (POST api/v1/notifications).

Here is what I got:
1) When both devices are logged in, both devices will get a push notification. (as expected)
2) When one of the two devices is logged out, the other one will get a push notification (as expected) HOWEVER here is what I would get from the api as a response: `{"id":"some-id","recipients":1,"external_id":null}` (I would have expected getting something like`{"id": "some-id","recipients": 1,"external_id": null,"errors": {"invalid_external_user_ids": ["id-of-the-logged-out-user"]}}`)

3) When both devices are logged out, both devices would not get any notification (as expected) and there api would return `{"id":"","recipients":0,"errors":["All included players are not subscribed"]}` (also expected)

I would have expected either getting 2 recipients even when both called removeExternalUserId (meaning that removeExternalUserId does not unsub the device) OR getting the response I mentioned inside 2).

FYI: I am using `"react-native-onesignal": "^3.9.0"`.

**Comment by rgomezp** [comment]
> When one of the two devices is logged out, the other one will get a push notification (as expected) HOWEVER here is what I would get from the api as a response: {"id":"some-id","recipients":1,"external_id":null} (I would have expected getting something like{"id": "some-id","recipients": 1,"external_id": null,"errors": {"invalid_external_user_ids": ["id-of-the-logged-out-user"]}})

That's not how it works. We link the records via the OneSignal "playerId" (user id) and external id. You can have multiple player ids linked with a single player id. If one gets "unlinked," we don't hold on to this previous relation.

> I would have expected either getting 2 recipients even when both called removeExternalUserId (meaning that removeExternalUserId does not unsub the device) OR getting the response I mentioned inside 2).

If you are targeting the devices via external id only, then yes - removing the links will essentially prevent those devices from being targeted.

Perhaps you can provide more context as to what you're trying to do?

If you want to, you can target devices via the player id only and not get mixed up with external ids.

**Comment by markoj3s** [comment]
Hi @rgomezp !

Sorry seems like I was not very clear.

What I was trying to do, is to know which **external_user_id** got its push notification delivered based on the **invalid_external_user_ids**.
Because the error message was **All included players are not subscribed**, I thought that, if only one of the users didn't get its push notification, OneSignal would return the unknown **external_user_ids** (and if none of the users got the notification, then I would get **All included players are not subscribed**).
So basically the fact that they provide the recipient count and let us know when none of them received a notification got me confused and made me think that the api would normally return the list of the unknown external_user_ids.
But since last time, I discussed with OneSignal support staff about this issue, and it seems like there are indeed no way to know who got a push notification that way.

**Comment by rgomezp** [comment]
Correct. There is no way to provide the list of unknown external user ids. Thank you for your question.

Enjoy!
